src/components/Rating/Rating.tsx

```diff
@@ -0,0 +1,330 @@
+import React, { Component } from 'react'
+import CSS from "csstype";
+
+export interface propTypes {
+  className?: string;
+  edit?: boolean;
+  half?: boolean;
+  value: number;
+  count: number;
+  char?: string;
+  size: number;
+  color1?: string;
+  color2?: string;
+  onChange: any;
+  labels: string[];
+  labelIndex: number;
+}
+
+interface stateTypes {
+  uniqueness?: string;
+  value: number;
+  stars: {active: boolean}[];
+  halfStar: {
+    at?: number;
```

Comment:
The `at` looks slightly confusing to me. Does it mean the current hovering index of star?

Reply:
done